src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -129,6 +136,95 @@ public void workspaceAuthz(AuthenticatedUserRequest userReq, UUID workspaceId, S
               userReq.getEmail(), action, workspaceId.toString()));
   }
 
+  /**
+   * Wrapper around Sam client to grant a role to the provided user.
+   *
+   * <p>This operation is only available to MC_TERRA stage workspaces, as Rawls manages permissions
+   * directly on other workspaces.
+   *
+   * @param workspaceId The workspace this operation takes place in
+   * @param userReq Credentials of the user requesting this operation. Only owners have permission
+   *     to modify roles in a workspace.
+   * @param role The role being granted.
+   * @param email The user being granted a role.
+   */
+  public void grantWorkspaceRole(
```

Comment:
The grant and the revoke methods are nearly identical, is there a way to DRY it up?
One idea: factoring out stage checking and throwing into a method.

Reply:
That gets used a few other places as well, moved it to a common method

Reply:
Nice - I like the name